virttest/qemu_storage.py

```diff
@@ -151,6 +152,33 @@ def get_image_repr(image, params, root_dir, representation=None):
     return func(image, params, root_dir)
 
 
+def create_temporary_image_snapshot(params, root_dir, vm_name, image_name):
+    """
+    Create temporary snapshot for image.
+
+    :param params: The image parameters.
+    :type params: dict
+    :param root_dir: Base directory for relative filenames.
+    :type root_dir: str
+    :param vm_name: The name of the corresponding VM.
+    :type vm_name: str
+    :param image_name: The image name.
+    :type image_name: str
+    :return: The filename of the temporary snapshot image.
+    :rtype: str
+    """
+    base_img_filename = storage.get_image_filename(params, root_dir)
+    img_format = params.get('image_format')
+    sn_name = 'vl_%s_%s' % (vm_name, image_name)
+    img_sn_filename = os.path.join(data_dir.get_tmp_dir(), sn_name)
+    qemu_img_bin = utils_misc.get_qemu_img_binary(params)
+    cmd = ' '.join((qemu_img_bin, 'create', '-f', img_format,
```

Comment:
we also need to given option '-F backing_fmt'.

Reply:
backing file maybe raw, qcow2 or luks.

Reply:
Thanks, will update.